bin/publish.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -e
+#
+# set version and publish to registry
+#
+
+# pipelines env
+branch=${BITBUCKET_BRANCH?}
+repoSlug=${BITBUCKET_REPO_SLUG?}
+
+# read current version
+version=$(npm show $repoSlug version)
+major="$(echo $version | cut -d'.' -f1)"
+minor="$(echo $version | cut -d'.' -f2)"
+patch="$(echo $version | cut -d'.' -f3 | cut -d'-' -f1)"
+
+# increment patch version
+patch="$(($patch + 1))"
+newVersion="$major.$minor.$patch"
+
+# set prerelease suffix for non-master builds
+if [ $branch != "master" ]; then
```

Comment:
I found that having default branch name configurable very useful for support of different versions. therefore we could make this configurable:

	function branchType () {
	  for b in ${DEFAULT_BRANCHES:-master}; do
	    [[ "$b" == "$1" ]] && echo default && return;
	  done
	  echo feature
	}

	# example usage
	case `branchType $branch` in
	"default")
	  echo default;
	  ;;
	*)
	  echo not default;
	  ;;
	esac

`DEFAULT_BRANCHES` could then be defined in pipelines settings to be `"master 1.x"` and builds from `master` and `1.x` would both be treated as release branches